erc20.go

```diff
@@ -78,17 +78,14 @@ func (p *Plugin) InitPlugin(metadata *persist.Metadata, bucket *bolt.Bucket, sto
 			Version: pluginDBVersion,
 			Header:  pluginDBHeader,
 		}
-	} else if metadata.Version != pluginDBVersion {
+	} else if metadata.Version != pluginDBVersion || metadata.Header != pluginDBHeader {
 		return persist.Metadata{}, errors.New("There is only 1 version of this plugin, version mismatch")
 	}
 	return *metadata, nil
 }
 
 // ApplyBlock applies a block's erc20 transaction to their buckets.
 func (p *Plugin) ApplyBlock(block rivinetypes.Block, height rivinetypes.BlockHeight, bucket *persist.LazyBoltBucket) error {
-	if bucket == nil {
```

Comment:
So the answer to my question was...? Did you re-check it?

Reply:
yes a nil bucket is never passed see `modules/consensus/plugin.go`